tests/instrumentation/transactions_store_tests.py

```diff
@@ -128,3 +130,29 @@ def test_get_transaction_clear():
     t = requests_store.begin_transaction("test")
     assert t == get_transaction(clear=True)
     assert get_transaction() is None
+
+
+def test_tag_transaction():
+    requests_store = TransactionsStore(lambda: [], 99999)
+    t = requests_store.begin_transaction("test")
+    elasticapm.tag(foo='bar')
+    requests_store.end_transaction(200, 'test')
+
+    assert t._tags == {'foo': 'bar'}
+    transaction_dict = t.to_dict()
+    assert transaction_dict['context']['tags'] == {'foo': 'bar'}
+
+
+def test_tag_while_no_transaction(caplog):
+    elasticapm.tag(foo='bar')
+    record = caplog.records[0]
+    assert record.levelno == logging.WARNING
+    assert 'foo' in record.args
+
+
+def test_tag_with_non_string_value():
+    requests_store = TransactionsStore(lambda: [], 99999)
+    t = requests_store.begin_transaction("test")
+    elasticapm.tag(foo=1)
+    requests_store.end_transaction(200, 'test')
+    assert t._tags == {'foo': '1'}
```

Comment:
you could add a test that checks the string pattern.

Reply:
I had such a test in #27. But because this API enforces stricter limits to the tag name than our regex (by virtue of being set as a function keyword), there is no way to use a tag name that would fail our regex.

Reply:
I remember, was just wondering, since you still validate against the regex.

Reply:
Yup, I thought about removing the regex check, but decided to leave it there a) for documentation purposes, and b) if we ever change the API